referenceframe/frame.go

```diff
@@ -124,10 +124,44 @@ type Frame interface {
 	json.Marshaler
 }
 
+// baseFrame contains all the data and methods common to all frames, notably it does not implement the Frame interface itself.
+type baseFrame struct {
+	name   string
+	limits []Limit
+}
+
+// Name returns the name of the referenceframe.
+func (bf *baseFrame) Name() string {
+	return bf.name
+}
+
+// DoF will return a slice with length equal to the number of joints/degrees of freedom.
+func (bf *baseFrame) DoF() []Limit {
+	return bf.limits
+}
+
+// validInputs checks whether the given array of joint positions violates any joint limits.
+func (bf *baseFrame) validInputs(inputs []Input) error {
```

Comment:
For clarity, how are joints without limits implemented here?

Reply:
`math.Inf(1)` and `math.Inf(-1)` are valid floats in Go, and are the defaults if the user does not add limits in their inputs.

Reply:
Frames that don't have limits (for example `staticFrame`) have no use for this function, as it is just meant to be a helper function to abstract away limit checking for frames that need it

Reply:
@biotinker Excellent